lib/git_local/repository.rb

```diff
@@ -48,10 +49,12 @@ def new_commit_on_remote?
       head = popened_io.read.chomp.split("\n").last
       Process.wait(popened_io.pid)
 
-      popened_io = IO.popen("(cd #{path} && git remote update && git rev-parse origin/#{branch})")
+      out = []
+      popened_io = IO.popen("(cd #{path} && git remote update && git rev-parse origin/#{branch}) 2>&1")
+      popened_io.each { |line| out << line.chomp }
       remote = popened_io.read.chomp
       Process.wait(popened_io.pid)
-      raise NotFound unless $?.to_i == 0
+      raise GitLocal::Repository::NotFound.new.exception(out.join(' ')) unless $?.to_i == 0
```

Comment:
This namespacing should be unnecessary: 

`raise NotFound.new.exception(out.join(' ')) unless $?.to_i == 0` should work